### ISSUE TITLE
cdk bootstrap: Error: Cannot find module './cdk.ts'

### ISSUE THREAD

**Issue by btakita** [aws/aws-cdk#24020]
### Describe the bug

Running:
```
cdk bootstrap
```
results in an error:
```
Error: Cannot find module './cdk.ts'
Require stack:
- /home/brian/work/cogov/cogov-dev/apps/cdk/bin/imaginaryUncacheableRequireResolveScript
    at Module._resolveFilename (node:internal/modules/cjs/loader:1039:15)
    at Function.resolve (node:internal/modules/cjs/helpers:109:19)
    at requireResolveNonCached (/home/brian/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/dist/bin.js:549:16)
    at getProjectSearchDir (/home/brian/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/dist/bin.js:519:40)
    at phase3 (/home/brian/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/dist/bin.js:267:27)
    at bootstrap (/home/brian/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/dist/bin.js:47:30)
    at main (/home/brian/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/dist/bin.js:33:12)
    at Object.<anonymous> (/home/brian/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/dist/bin.js:579:5)
    at Module._compile (node:internal/modules/cjs/loader:1218:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1272:10) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/brian/work/cogov/cogov-dev/apps/cdk/bin/imaginaryUncacheableRequireResolveScript'
  ]
}
```

### Expected Behavior

There should be no error

### Current Behavior

```
Error: Cannot find module './cdk.ts'
Require stack:
- /home/brian/work/cogov/cogov-dev/apps/cdk/bin/imaginaryUncacheableRequireResolveScript
    at Module._resolveFilename (node:internal/modules/cjs/loader:1039:15)
    at Function.resolve (node:internal/modules/cjs/helpers:109:19)
    at requireResolveNonCached (/home/brian/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/dist/bin.js:549:16)
    at getProjectSearchDir (/home/brian/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/dist/bin.js:519:40)
    at phase3 (/home/brian/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/dist/bin.js:267:27)
    at bootstrap (/home/brian/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/dist/bin.js:47:30)
    at main (/home/brian/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/dist/bin.js:33:12)
    at Object.<anonymous> (/home/brian/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/dist/bin.js:579:5)
    at Module._compile (node:internal/modules/cjs/loader:1218:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1272:10) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/brian/work/cogov/cogov-dev/apps/cdk/bin/imaginaryUncacheableRequireResolveScript'
  ]
}
```

### Reproduction Steps

Run `cdk bootstrap`

### CDK CLI Version

2.63.2 (build e08e34a)

### Node.js Version

18.13.0

### OS

Linux

### Language

Typescript

**Comment by btakita** [comment]
The cause of this error was an incorrect setting in `cdk.json` `app`.

**Comment by AllanOricil** [comment]
@btakita what was the incorrect setting?

**Comment by btakita** [comment]
The `app` setting was pointing to a missing file.

**Comment by AllanOricil** [comment]
> The `app` setting was pointing to a missing file.

I had this error in codepipelines, in aws.

I have the following stages define: source -> build -> deploy

After building everything, in the next stage, I ran `cdk deploy --all --ci --request-approval never` and got this same error. Before running it, I added some `ls` and `cat` to check if the file specified in `app`, in cdk.json, was missing, and it wasn't. I concluded that this isn't the only root cause of this problem.
